crates/assists/src/handlers/extract_struct_from_enum_variant.rs

```diff
@@ -204,27 +203,33 @@ fn update_reference(
     variant_hir_name: &Name,
     visited_modules_set: &mut FxHashSet<Module>,
 ) -> Option<()> {
-    let path_expr: ast::PathExpr = find_node_at_offset::<ast::PathExpr>(
-        source_file.syntax(),
-        reference.file_range.range.start(),
-    )?;
-    let call = path_expr.syntax().parent().and_then(ast::CallExpr::cast)?;
-    let list = call.arg_list()?;
-    let segment = path_expr.path()?.segment()?;
-    let module = ctx.sema.scope(&path_expr.syntax()).module()?;
+    let offset = reference.file_range.range.start();
+    let (segment, expr) = if let Some(path_expr) =
+        find_node_at_offset::<ast::PathExpr>(source_file.syntax(), offset)
+    {
+        // tuple variant
+        (path_expr.path()?.segment()?, path_expr.syntax().parent()?.clone())
+    } else if let Some(record_expr) =
+        find_node_at_offset::<ast::RecordExpr>(source_file.syntax(), offset)
+    {
+        // record variant
+        (record_expr.path()?.segment()?, record_expr.syntax().clone())
+    } else {
+        return None;
+    };
```

Comment:
I wonder if this'll work correctly for ast::RecordExpr nested into an `ast::CallExpr` for a different tuple struct?

Reply:
You mean something like this?
```rust
enum A { <|>One { a: u32, b: u32 } }

struct B(A);

fn foo() {
    let _ = B(A::One { a: 1, b: 2 });
}
```
This seems to work properly as I get
```rust
struct One{ pub a: u32, pub b: u32 }

enum A { One(One) }

struct B(A);

fn foo() {
    let _ = B(A::One(One { a: 1, b: 2 }));
}
```
which seems correct to me. Though I would've expected this to fail as well now that you say it.
Will definitely add this as a test if anything.